### PR TITLE
ref(js): Convert GlobalSelectionLink to a FC

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/components/globalSelectionLink.tsx [hunk 1]
```diff
@@ -32,49 +32,47 @@ type Props = WithRouterProps & {
...
-      const routerProps = hasGlobalQuery
-        ? {...this.props, to: toWithGlobalQuery}
-        : {...this.props, to};
+    const routerProps = hasGlobalQuery
+      ? {...props, to: toWithGlobalQuery}
+      : {...props, to};
 
-      return <Link {...routerProps}>{this.props.children}</Link>;
-    }
+    return <Link {...routerProps}>{props.children}</Link>;
```
Comment (priscilawebdev) [inline static/app/components/globalSelectionLink.tsx:54]:
```suggestion
    return <Link {...routerProps}>{children}</Link>;
```

File: static/app/components/globalSelectionLink.tsx [hunk 2]
```diff
@@ -32,49 +32,47 @@ type Props = WithRouterProps & {
  *
  * Falls back to <a> if there is no router present.
  */
-class GlobalSelectionLink extends React.Component<Props> {
-  render() {
-    const {location, to} = this.props;
+function GlobalSelectionLink(props: Props) {
+  const {location, to} = props;
```
Comment (priscilawebdev) [inline static/app/components/globalSelectionLink.tsx:36]:
```suggestion
  const {location, to, children} = props;
```

File: static/app/components/globalSelectionLink.tsx [hunk 3]
```diff
@@ -7,6 +6,7 @@ import Link from 'sentry/components/links/link';
 import {extractSelectionParameters} from 'sentry/components/organizations/globalSelectionHeader/utils';
 
 type Props = WithRouterProps & {
+  children?: React.ReactChildren;
```
Comment (priscilawebdev) [inline static/app/components/globalSelectionLink.tsx:9]: should we replace `React.ReactChildren` with `React.ReactNode` here?

Comment (evanpurkhiser) [reply static/app/components/globalSelectionLink.tsx]: Yeah.. it's not totally clear to me when to use which haha

File: static/app/components/globalSelectionLink.tsx (hunk 2 above)
Comment (evanpurkhiser) [reply static/app/components/globalSelectionLink.tsx:36]: I updated the PR and just let children spread into the element

File: static/app/components/globalSelectionLink.tsx [hunk 4]
```diff
@@ -7,6 +6,7 @@ import Link from 'sentry/components/links/link';
 import {extractSelectionParameters} from 'sentry/components/organizations/globalSelectionHeader/utils';
 
 type Props = WithRouterProps & {
+  children?: React.ReactNode;
```
Comment (priscilawebdev) [inline static/app/components/globalSelectionLink.tsx:9]: since this prop is optional, can we move it to the bottom of the object? :)

**Review by priscilawebdev — APPROVED** [review]
left one suggestion but it looks good

File: static/app/components/globalSelectionLink.tsx (hunk 4 above)
Comment (evanpurkhiser) [reply static/app/components/globalSelectionLink.tsx:9]: will move it in a follow up!

Comment (evanpurkhiser) [reply static/app/components/globalSelectionLink.tsx:9]: https://github.com/getsentry/sentry/pull/30711

**Pull request merged**